docs-src/design-decisions.md

```diff
@@ -0,0 +1,42 @@
+# Design Decisions
+
+Throughout the design of these APIs some significant design decisions have been
+made. This provides some background on the factors considered when making these
+decisions.
+
+## GatewayClass Scope
+
+### Problem
+There were a number of compelling reasons to make GatewayClass a
+namespace-scoped resource. 
+
+### Advantages of a Namespace-Scoped GatewayClass
+* Less potential for naming conflicts among different implementations.
+* Better modeling for implementations with multiple GatewayClass resources
+  (including some potential multi-tenant scenarios using Cluster API).
+* Easier to provide simple and secure defaults (only allow Gateways and Routes
+  in the same namespace by default).
+* GatewayClass parameters can live in the same namespace.
+* More RBAC flexibility (can allow users to create GatewayClasses in select
+  namespaces only or restrict visibility of GatewayClasses in some namespaces).
+
+### Advantages of a Cluster-Scoped GatewayClass
+* Consistent with StorageClass and IngressClass resources.
+* Discussions with sig-storage showed that StorageClass had worked well as a
+  cluster-scoped resource and that they were continuing to follow that pattern
+  for new Class resources.
+* Follows general pattern that resources managed by infrastructure provider
```

Comment:
This pattern does not, to my knowledge, account for the concept of a management cluster.

Reply:
I think you're right. Is there any updated guidance here we can apply? Since this pattern has already been applied so widely in k/k, has anyone found a good way to work around or solve these potential issues with management clusters?

Reply:
Unfortunately I don’t. It’s why I’ve been tagging colleagues at VMware and sharing this issue. I hate that this is the proving ground for this, but this is a really interesting and broad topic on the class pattern in the world of management clusters. 

I already pinged Vince Priganano and the rest of the cluster API team working at VMware to take a look. I know you pinged Saad. I’m really interested to see what people think.